src/main/java/com/google/cloud/genomics/dataflow/pipelines/ShardedBAMWriting.java

```diff
@@ -456,22 +537,100 @@ static String combineShards(ShardedBAMWritingOptions options, String dest,
       os.write(BlockCompressedStreamConstants.EMPTY_GZIP_BLOCK);
       os.close();
       sortedShardsNames.add(eofFileName);
-      // list of files to concatenate
-      ArrayList<SourceObjects> sourceObjects = new ArrayList<SourceObjects>();
-      for (String shard : sortedShardsNames) {
-          final GcsPath shardPath = GcsPath.fromUri(shard);
-          LOG.info("Adding object " + shardPath);
-          sourceObjects.add( new SourceObjects().setName(shardPath.getObject()) );
+      
+      int stageNumber = 0;
+      while (sortedShardsNames.size() > 32) {
```

Comment:
nit: use a constant and add a comment as to why combine must happen in groups of 32.